### PR TITLE
Death Dismissal Part 4 Frontend

### PR DESCRIPTION
**Pull request by lomky**
Connects #12211 

Continues work from #12529; #12535; #12543 

:red_circle: MERGE BLOCKED by #12535 & #12543. Contains redundant code until those are merged.

### Description
Adds the front end React components for the Death Dismissal ability.

### Acceptance Criteria
- [x] Code compiles
- Components cannot be accessed until the next PR, where they will be exercised!

A part of a larger set of PRs.
#12529 #12535 #12543 `#12544` #12545

### CODE REVIEW

File: spec/models/legacy_appeal_spec.rb [hunk 1]
```diff
@@ -2393,6 +2393,53 @@
...
+      end
+    end
+
+    context "if there are Caseflow tasks on the legacy appeal" do
+      context "multiple open caseflow tasks" do
+        it "cancels all the open tasks" do
+          appeal2.cancel_open_caseflow_tasks!
+
+          expect(appeal2.tasks.open.count).to eq(0)
+          expect(appeal2.tasks.closed.count).to eq(3)
```
Comment (yoomlam) [inline spec/models/legacy_appeal_spec.rb]: Do `Constants.TASK_STATUSES.cancelled` tasks for legacy appeals appear in the `appeal2.tasks.closed` list? Should it?

File: spec/models/legacy_appeal_spec.rb [hunk 2]
```diff
@@ -2393,6 +2393,53 @@
...
+            task = appeal2.tasks.first
+            task.update(instructions: ["Existing instructions"])
+            appeal2.cancel_open_caseflow_tasks!
+            expect(task.reload.instructions)
+              .to eq(["Existing instructions", "Task cancelled due to death dismissal"])
+          end
+        end
+      end
+
+      context "open and closed caseflow tasks" do
```
Comment (yoomlam) [inline spec/models/legacy_appeal_spec.rb]:
```suggestion
      context "when there are open and closed caseflow tasks" do
```

File: client/app/queue/components/DeathDismissalModal.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+  submit = () => {
+    return this.submitDeathDismissal();
+  };
+
+  submitDeathDismissal = () => {
+    const { appeal } = this.props;
+    const payload = {
+      data: {}
+    };
```
Comment (yoomlam) [inline client/app/queue/components/DeathDismissalModal.jsx]: Does it still work if `payload` was null?

File: client/app/queue/components/DeathDismissalModal.jsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      appeal.veteranFullName
+    );
+    const detail = (
+      <p>
+        {COPY.DEATH_DISMISSAL_SUCCESS_DETAIL}
+      </p>
+    );
+
+    return { title,
+      detail };
```
Comment (yoomlam) [inline client/app/queue/components/DeathDismissalModal.jsx]: (Nitpick) Combine into single line?

File: client/COPY.json [hunk 5]
```diff
@@ -467,6 +467,12 @@
   "UNAUTHORIZED_PAGE_ACCESS_MESSAGE": "You aren't authorized to use this part of Caseflow yet.",
   "CASE_DETAILS_LOADING_FAILURE_TITLE": "Unable to load this case",
 
+  "DEATH_DISMISSAL_MODAL_TITLE": "FNOD Death Dismissal",
```
[the remark this answers is not in the document]
Comment (lomky) [reply client/COPY.json]: Nope, the `S` in SFNOD is a VACOLS table indicator. The users know this as an FNOD

File: spec/models/legacy_appeal_spec.rb (hunk 1 above)
Comment (lomky) [reply spec/models/legacy_appeal_spec.rb]: Yes and yes!

https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/task.rb#L54

**Pull request closed without merging**